ssl/ssl_asn1.c

```diff
@@ -316,6 +330,18 @@ SSL_SESSION *d2i_SSL_SESSION(SSL_SESSION **a, const unsigned char **pp,
     ret->peer = as->peer;
     as->peer = NULL;
 
+    EVP_PKEY_free(ret->peer_rpk);
+    ret->peer_rpk = NULL;
+    if (as->peer_rpk != NULL) {
+        const unsigned char *data = as->peer_rpk->data;
+
+        /*
+         * |data| is incremented; we don't want to lose original ptr
+         * No libctx/propq available, so use non-ex version
+         */
+        ret->peer_rpk = d2i_PUBKEY(NULL, &data, as->peer_rpk->length);
```

Comment:
I usually check that the entire/expected length of the input is consumed by d2i. If decoding fails here, perhaps also return an error?

Reply:
@mattcaswell Should failure to decode the RPK be a hard-failure to decode the session?
(I'm approving the PR with this question left to you and any other reviewers).

Reply:
I'm inclined to agree with @vdukhovni here